src/SFML/Graphics/ImageLoader.hpp

```diff
@@ -121,25 +139,6 @@ class ImageLoader
                            std::vector<std::uint8_t>&       output,
                            const std::vector<std::uint8_t>& pixels,
                            const Vector2u&                  size);
-
-private:
-    ////////////////////////////////////////////////////////////
-    /// \brief Default constructor
-    ///
-    ////////////////////////////////////////////////////////////
-    ImageLoader();
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Deleted copy constructor
-    ///
-    ////////////////////////////////////////////////////////////
-    ImageLoader(const ImageLoader&) = delete;
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Deleted copy assignment
-    ///
-    ////////////////////////////////////////////////////////////
-    ImageLoader& operator=(const ImageLoader&) = delete;
```

Comment:
Did clang-tidy instruct you to move these to the header file?

Reply:
'
[1m/home/norme/Projects/SFML/src/SFML/Graphics/ImageLoader.hpp:136:5: [0m[0;1;31merror: [0m[1mdeleted member function should be public [modernize-use-equals-delete,-warnings-as-errors][0m
    ImageLoader(const ImageLoader&) = delete;
[0;1;32m    ^
[0m[1m/home/norme/Projects/SFML/src/SFML/Graphics/ImageLoader.hpp:142:18: [0m[0;1;31merror: [0m[1mdeleted member function should be public [modernize-use-equals-delete,-warnings-as-errors][0m
    ImageLoader& operator=(const ImageLoader&) = delete;
[0;1;32m                 ^
'

Reply:
I was confusing this with other files where code did move from .cpp files to .hpp files. In some cases the definition of those special member functions was kept in the .cpp for a reason so I'd prefer we not change that in this PR. The changes in sf::String demonstrate what I'm talking about.

Reply:
> I was confusing this with other files where code did move from .cpp files to .hpp files. In some cases the definition of those special member functions was kept in the .cpp for a reason so I'd prefer we not change that in this PR. The changes in sf::String demonstrate what I'm talking about.

Declaring `String()` in the header, then defining `String() = default` in the source seems redundant; is that what you would prefer?

Reply:
Yes it seems odd and I'm not sure it's my personal ideal but for now I'd rather not go moving those definitions around so we can keep the scope of this PR manageable.